### PR TITLE
Fix in-progress crafting of liquid items to have liquid behaviour

### LINKED ISSUE #35863: While crafting water to clean water , if you stop and not finish you can drop liquid and pick it up again
# Describe the bug

When you start crafting clean water from water , you start the progress and the item changes to "crafting progress (0%) clean water" , if you do not finish you can drop the current liquid that is being craft able at the ground without it being dirty and without using a container you can pick it up again.

# Steps To Reproduce

1. Get (Pot , Water , Fire)
2. Try crafting clean water , BUT don't finish crafting it completely 
3. you can store it anywhere without it getting dirty or without using any container , yet you cant use it.

# Expected behavior

should behave the same way as liquids do even if its in crafting state.

# Screenshots

water1
water2
water3

# Versions and configuration

- OS:

>  Windows 10

- Game Version:

> 0.D-9827-g4b589b7 (tiles)

- Graphics version:

> Tiles

- Mods loaded:
mods

# Additional context

i am annoying ik , but bugs are bugs.

### PR DESCRIPTION
**Pull request by conghienquoc**
#### Summary
```SUMMARY: Bugfixes "Fix in-progress crafting of liquid items to have liquid behaviour"```

#### Purpose of change

Fixes #35863 - if you do not finish crafting liquid, you can drop it and pick it up from the ground.

#### Describe the solution
1. In item constructor for crafting items, checks for if the resulting item is of LIQUID phase to determine if the current in-progress crafting item shoudl be of LIQUID phase.
2. In pick_up_one, add a check for crafting item in LIQUID phase.

#### Describe alternatives you've considered
If modifying the phase of the recipe item is allowed (it's currently a const pointer), then the added code in pick_up_one is not necessary.

#### Testing
In-game testing through the debug menu.

### CODE REVIEW

File: src/pickup.cpp [hunk 1]
```diff
@@ -272,6 +272,8 @@ bool pick_one_up( item_location &loc, int quantity, bool &got_water, bool &offer
         }
     } else if( newit.made_of_from_type( LIQUID ) && !newit.is_frozen_liquid() ) {
         got_water = true;
+    } else if( newit.is_craft() && newit.made_of( LIQUID )) {
+        got_water = true; //Handles in progress liquid craft
```
Comment (DaviBones) [inline src/pickup.cpp]: These do the same thing, so they can just be combined into one check, e.g.:
```cpp
    } else if( ( newit.made_of_from_type( LIQUID ) && !newit.is_frozen_liquid() ) ||
               ( newit.is_craft() && newit.made_of( LIQUID ) ) ) {
        got_water = true;
```

**Comment by ZhilkinSerg** [comment]
You need to astyle your changes (run `make astyle` or see https://github.com/CleverRaven/Cataclysm-DDA/pull/24006 if you are using VS).

File: src/item.cpp [hunk 2]
```diff
@@ -320,6 +320,11 @@ item::item( const recipe *rec, int qty, std::list<item> items, std::vector<item_
             }
         }
     }
+
+    // in craft of liquid item should have liquid attribute
+    if( find_type(making->result())->phase == LIQUID ) {
+        current_phase = static_cast<phase_id>( 2 );
```
Comment (BevapDin) [inline src/item.cpp]:
```suggestion
        current_phase = LIQUID;
```

**Comment by kevingranade** [comment]
I'm not sure if this does what we want, the recipe acts as a liquid even though any liquid recipes are going to include a container.
For example in the case of the clean water recipe, you need the water, a heat source, and a pot, so the more reasonable assumption is that the intermediate form is a pot full of water rather than treating it as some kind of liquid.

**Comment by kevingranade** [comment]
I think the real issue here is mislabelling, an in-progress "clean water" craft is going to be called "in progress clean water", but in reality it's "a pot of in-progress clean water".  Basically the behavior is fine, but the description could be clearer about what's happening.

**Comment by Feuerhai** [comment]
But the "in progress clean water" doesn't have the pot included in the item itself, so you can just store it in a cupboard or something while taking the pot with you.

**Comment by ZhilkinSerg** [comment]
> 
> 
> But the "in progress clean water" doesn't have the pot included in the item itself, so you can just store it in a cupboard or something while taking the pot with you.

Container should be part of incomplete craft and not be available for other activities.

**Pull request closed without merging**